tests/tensorflow_autolog/test_tensorflow2_autolog.py

```diff
@@ -458,6 +458,39 @@ def test_tf_keras_autolog_non_early_stop_callback_no_log(tf_keras_random_data_ru
     assert len(metric_history) == num_of_epochs
 
 
+@pytest.mark.parametrize("fit_variant", ["fit", "fit_generator"])
+def test_tf_keras_autolog_does_not_mutate_original_callbacks_list(
```

Comment:
Just making sure, did this test fail without your change above?

Reply:
Oh yeah, big time. It still fails against TF 2.0.4 - will need to look into that before merging

Reply:
Fixed the 2.0.4 issue (needed to apply similar fix logic to our patch for `fit_generator()`, which is only relevant for TF < 2.1.0 (e.g. 2.0.4)